### DISCUSSION TITLE
For Cloud Functions 2nd gen

### DISCUSSION THREAD

**Discussion by akehoyayoi** [firebase/firebase-ios-sdk#10198] [Q&A] [answered]
I cloud not find the milestone for adapting cloud functions 2nd gen as far as I confirm.
Do anyone know which firebase-ios-sdk has a plan for adapting cf 2nd gen or not?

**Comment by inlined** [comment, marked as answer]
**EDIT**:  Cloud Functions for Firebase 2nd Gen does not need any special handling anymore for callable functions to work in the iOS SDK. This restriction was removed as part of our GA announcement at Google I/O 2023.

All changes that are needed for the iOS SDK to work have already been made.

Eventually, the code you use for a v1 callable function will just work with v2 callable functions seamlessly. For now, however, we don't yet have cloudfunctions.net URLs for v2 functions, which is what the SDK normally depends on. Instead, you'll temporarily need to first deploy your function, then copy the URL you get from the deploy logs and initialize your callable function with the [URL based constructor](https://firebase.google.com/docs/reference/swift/firebasefunctions/api/reference/Classes/Functions#/c:@M@FirebaseFunctions@objc(cs)FIRFunctions(im)HTTPSCallableWithURL:)

**Reply by akehoyayoi** [reply]
Thanks for your reply! Understood!

**Reply by walkingbrad** [reply]
Has this been improved at all yet? Or where can I tune in to find out when that happens?

**Reply by inlined** [reply]
Added an edit on the bug that no workarounds are necessary anymore.
